Fix `Tucan.Data.column_types/1` when data is empty

jechol opened the pull request.

pnezis commented:
Thanks @jechol, can you add a test as well?

Something like this under the new describe block would be fine:

```elixir
test "with empty data" do
   vl = Tucan.new([])

   assert get_in(vl.spec, ["__tucan__", "types"]) == %{}
end
```

pnezis commented on lib/tucan/data.ex:
```diff
@@ -8,15 +8,15 @@ defmodule Tucan.Data do
       Enum.zip_with(columns, types, fn column, type -> {to_string(column), type} end) |> Map.new()
     else
-      _ -> nil
+      _ -> %{}
```
You will also have to update the `Tucan.Data` tests

jechol commented:
Thanks, @pnezis! I’ve added the test for empty data and updated the Tucan.Data tests as requested. Let me know if there’s anything else I should adjust. 🚀
